src/javascript/ContentEditor/actions/contenteditor/copyLanguage/copyLanguage.utils.js

```diff
@@ -17,27 +18,24 @@ export function getFullLanguageName(languages, id) {
  * @param formAndData form and values of a node
  * @returns The internationalized field and their values
  */
-// eslint-disable-next-line no-warning-comments
-// TODO: https://jira.jahia.org/browse/TECH-301 we should use the Edit.adapter.js getFieldValues, because some field require adaptation before injected in the form
 export function getI18nFieldAndValues(formAndData) {
     const {forms, jcr} = formAndData.data;
     let i18nFields = forms.editForm.sections
         .flatMap(section => section.fieldSets)
         .flatMap(fieldSet => fieldSet.fields)
         .filter(field => field.i18n)
-        .map(field => {
-            return {
-                name: field.name,
-                multiple: field.multiple
-            };
-        });
+        .map(field => ({...field, propertyName: field.name}));
+
+    const adaptedValues = i18nFields.map(field => getFieldValues(field, jcr?.result)).reduce((acc, current) => ({...acc, ...current}), {});
```

Comment:
Overal the method: getI18nFieldAndValues seem's a bit clunky, even before your modifications:
- we could adapt field on demand in the return statement to avoid adapting fields that could be filtered few lines later.
- I count 3 iterations on i18nFields, seem's a lot for what we need to do here. (2x `i18nFields.map`, 1x `i18nFields.find`) I think the algorythms could be simplified here and be more efficient and also more easy to read.

Consider:
In the return statement, when doing: `jcr.result.properties.filter` getting the associated field at the same time and adapt on the fly the value.